pkg/koordlet/statesinformer/container_informer.go

```diff
@@ -0,0 +1,247 @@
+package statesinformer
+
+import (
+	"context"
+	topov1alpha1 "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/apis/topology/v1alpha1"
+	slov1alpha1 "github.com/koordinator-sh/koordinator/apis/slo/v1alpha1"
+	"golang.org/x/time/rate"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/klog/v2"
+	"sync"
+	"time"
+
+	"go.uber.org/atomic"
+	runtimeapi "k8s.io/cri-api/pkg/apis/runtime/v1alpha2"
+
+	"github.com/koordinator-sh/koordinator/pkg/koordlet/pleg"
+	"github.com/koordinator-sh/koordinator/pkg/koordlet/util/runtime"
+	"github.com/koordinator-sh/koordinator/pkg/koordlet/util/runtime/handler"
+	"github.com/koordinator-sh/koordinator/pkg/koordlet/util/system"
+)
+
+const (
+	ContainerInformerName informerName = "ContainerInformer"
+)
+
+type ContainerInformer interface {
+	Run(stopCh <-chan struct{}) error
+	HasSynced() bool
+
+	GetContainer(containerHashID string) *ContainerMeta
+	GetAllSandboxContainers() []*ContainerMeta
+	GeSandboxContainer(podUID string) *ContainerMeta
+	GetContainersByPodUID(podUID string) []*ContainerMeta
+
+	RegisterCallbacks(objType RegisterType, name, description string, callbackFn UpdateCbFn)
+}
+
+type createEvent struct {
+	containerID string
+	podID       string
+}
+
+type containerInformer struct {
+	config *Config
+
+	containerUpdatedTime time.Time
+	containerRWMutex     sync.RWMutex
+	containerMap         map[string]map[string]*ContainerMeta
+	containerHasSynced   *atomic.Bool
+
+	// use pleg to accelerate the efficiency of container update
+	pleg             pleg.Pleg
+	containerCreated chan createEvent
+
+	runtimeServiceClient runtimeapi.RuntimeServiceClient
+}
+
+func NewContainerInformer() *containerInformer {
+	p, err := pleg.NewPLEG(system.Conf.CgroupRootDir)
+	if err != nil {
+		klog.Fatalf("failed to create PLEG, %v", err)
+	}
+
+	informer := &containerInformer{
+		containerMap:       make(map[string]map[string]*ContainerMeta),
```

Comment:
divide into two maps may be easy understanding.
map[podUID][]*ContainersMeta
map[ContainerHashID]*ContainerMeta